### PR TITLE
METRON-1179: Make STATS_ADD to take a list

### PR DESCRIPTION
**Pull request by cestella**
## Contributor Comments
While we can add multiple values in `STATS_ADD` via `STATS_ADD(s, 1, 2, 3)`, but it'd be nice also to support `STATS_ADD(s, [1, 2, 3])`

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-analytics/metron-statistics/src/main/java/org/apache/metron/statistics/StellarStatisticsFunctions.java [hunk 1]
```diff
@@ -141,8 +141,21 @@ public Object apply(List<Object> args) {
 
       // add each of the numeric values
       for(int i=1; i<args.size(); i++) {
-        double value = convert(args.get(i), Double.class);
-        stats.addValue(value);
+        Object n = args.get(i);
+        if( n != null) {
+          if(n instanceof Iterable) {
+            for(Object num : (Iterable<Object>)n) {
+              if(num != null) {
```
Comment (ottobackwards) [inline metron-analytics/metron-statistics/src/main/java/org/apache/metron/statistics/StellarStatisticsFunctions.java]: Just a nit, but why 'double' and not Double?
Is there a reason to box?

File: metron-analytics/metron-statistics/README.md [hunk 2]
```diff
@@ -60,7 +60,7 @@ functions can be used from everywhere where Stellar is used.
   * Description: Adds one or more input values to those that are used to calculate the summary statistics.
   * Input:
     * stats - The Stellar statistics object.  If null, then a new one is initialized.
-    * value+ - One or more numbers to add
+    * value+ - One or more numbers to add. If one of objects is a list, it will iterate over the list and add each number.
```
Comment (ottobackwards) [inline metron-analytics/metron-statistics/README.md]: "One or more items to add.  Each item may be a number or a list of numbers.  If an item is a list,  each number in the list will be added"?

**Comment by ottobackwards** [comment]
Looks great.  Just a couple of nits, but otherwise LGTM +1.

File: metron-analytics/metron-statistics/src/main/java/org/apache/metron/statistics/StellarStatisticsFunctions.java (hunk 1 above)
Comment (cestella) [reply metron-analytics/metron-statistics/src/main/java/org/apache/metron/statistics/StellarStatisticsFunctions.java]: No good reason :)

**Comment by cestella** [comment]
Ok, this is updated as per your comments, @ottobackwards

**Comment by ottobackwards** [comment]
+1, thanks for the contribution and great work.

**Pull request closed without merging**